middleware.go

```diff
@@ -168,6 +171,30 @@ func (t BaseMiddleware) OrgSessionExpiry(orgid string) int64 {
 	return cachedVal.(int64)
 }
 
+func (t BaseMiddleware) UpdateRequestSession(r *http.Request) bool {
+	session := ctxGetSession(r)
+	token := ctxGetAuthToken(r)
+
+	if session == nil || token == "" {
+		return false
+	}
+
+	if !ctxSessionUpdateScheduled(r) {
+		return false
+	}
+
+	lifetime := session.Lifetime(t.Spec.SessionLifetime)
+	if err := t.Spec.SessionManager.UpdateSession(token, session, lifetime, false); err != nil {
+		return false
+	}
+
```

Comment:
I think we also need to update session in cache if `!config.Global().LocalSessionCache.DisableCacheSessionState`

Reply:
I actually believe it will be handled by itself. Session cache holds only pointers to session object, so if session itself modified, the cache should be too.

Reply:
when I was working on `UpdateSession` I found that we don't sent to session cache pointers, it is one of results of `SessionManager.SessionDetail`, `CheckSessionAndIdentityForValidKey` or `AuthManager.KeyAuthorised` which all returns just structs. it probably was done on purpose to avoid side effects of storing pointers and update cache explicitly (I think pointers eliminate updating cache items atomically)

Reply:
You are right, now I update the cache